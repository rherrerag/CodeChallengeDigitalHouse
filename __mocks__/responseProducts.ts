const responseOK = {
  data: [
    {
      createdAt: '2023-03-11T04:50:50.205Z',
      product: 'Recycled Plastic Tuna',
      points: 92984,
      image: 'https://loremflickr.com/640/480/technics',
      is_redemption: false,
      id: '2',
    },
    {
      createdAt: '2023-03-11T04:35:35.259Z',
      product: 'Fantastic Granite Bacon',
      points: 42416,
      image: 'https://loremflickr.com/640/480/technics',
      is_redemption: false,
      id: '3',
    },
    {
      createdAt: '2023-03-11T13:57:33.428Z',
      product: 'Fantastic Fresh Gloves',
      points: 23913,
      image: 'https://loremflickr.com/640/480/city',
      is_redemption: true,
      id: '4',
    },
    {
      createdAt: '2023-03-11T12:17:47.118Z',
      product: 'Rustic Rubber Bacon',
      points: 69814,
      image: 'https://loremflickr.com/640/480/people',
      is_redemption: true,
      id: '5',
    },
    {
      createdAt: '2023-02-04T07:48:16.249Z',
      product: 'Licensed Metal Salad',
      points: 93367,
      image: 'https://loremflickr.com/640/480/food',
      is_redemption: false,
      id: '12',
    },
  ],
  error: false,
  isCancel: false,
};

const responseError = {
  data: {message: 'Error', code: 'ERR'},
  error: true,
};

export {responseOK, responseError};
