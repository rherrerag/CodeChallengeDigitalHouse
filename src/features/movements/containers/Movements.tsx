import React, {useEffect, useState} from 'react';
import {FlatList, ListRenderItem, Dimensions} from 'react-native';
import {Box, VStack, HStack, Heading, Spinner} from 'native-base';
import {useSelector} from 'react-redux';
import {useAppDispatch} from '../../../common/hooks/hooks';
import {
  fetchMovementsData,
  setEarnedMovements,
  setRedeemedMovements,
} from '../reducers/movements.actions';
import {
  selectMovements,
  selectEarnedMovements,
  selectRedeemedMovements,
} from '../reducers/movements.selectors';
import ProductItem from '../components/ProductItem';
import {Movement} from '../interfaces/movement.interface';
import {
  EarnedTotalButton,
  RedeemedTotalButton,
  TotalButton,
  FlatListContainer,
  Subtitle,
  CardMonth,
  CardPointsTotal,
  FullContainer,
  CardContainer,
  CardPadding,
  WelcomeText,
  WelcomeName,
  ButtonsContainer,
} from './Movements.styled';
import labels from '../constants/labels';
import {SafeAreaView} from 'react-native-safe-area-context';
import {sumPoints} from '../utils/pointUtils';
import {filterByDate, monthNames} from '../../../common/utils/dateUtils';
import config from '../../../constants/config';

const Movements = () => {
  const dispatch = useAppDispatch();

  const [allMovements, setAllMovements] = useState(true);
  const [loading, setLoading] = useState(true);
  const [currentMovements, setCurrentMovements] = useState<Movement[]>([]);
  const movements = useSelector(selectMovements);
  const earnedMovements = useSelector(selectEarnedMovements);
  const redeemedMovements = useSelector(selectRedeemedMovements);

  const {height} = Dimensions.get('window');
  const styleHeight = {height: height / 2.5};

  const currentMonth = monthNames[Number(config.FILTER.MONTH) - 1];

  const totalPoints = sumPoints(earnedMovements) - sumPoints(redeemedMovements);
  const cardPoints =
    totalPoints.toLocaleString('es-MX') + ' ' + labels.CARD.POINTS;

  useEffect(() => {
    setLoading(true);
    dispatch(fetchMovementsData());
  }, []);

  useEffect(() => {
    if (movements.length) {
      const movementsFiltered = filterByDate(
        movements,
        config.FILTER.MONTH,
        config.FILTER.YEAR,
      );
      setCurrentMovements(movementsFiltered);
      dispatch(
        setEarnedMovements(
          movementsFiltered.filter(element => !element.is_redemption),
        ),
      );
      dispatch(
        setRedeemedMovements(
          movementsFiltered.filter(element => element.is_redemption),
        ),
      );
      setLoading(false);
    }
  }, [movements, dispatch]);

  const allMovementsPressed = () => {
    setCurrentMovements(
      filterByDate(movements, config.FILTER.MONTH, config.FILTER.YEAR),
    );
    setAllMovements(true);
  };

  const filterMovementsPressed = (flag: Boolean) => {
    let filteredMovements: Movement[] | null = [];
    flag
      ? (filteredMovements = earnedMovements)
      : (filteredMovements = redeemedMovements);
    setCurrentMovements(filteredMovements);
    setAllMovements(false);
  };

  const renderFilterButtons = () => {
    return (
      <Box>
        <HStack space={2} justifyContent="center">
          <EarnedTotalButton
            testID="earned-points-button"
            bgColor="primary.100"
            _text={{
              fontSize: 12,
              fontWeight: 800,
              color: 'white',
              lineHeight: 16,
            }}
            onPress={() => filterMovementsPressed(true)}>
            {labels.BUTTONS.EARNED}
          </EarnedTotalButton>
          <RedeemedTotalButton
            testID="redeemed-points-button"
            bgColor="primary.100"
            _text={{
              fontSize: 12,
              fontWeight: 800,
              color: 'white',
              lineHeight: 16,
            }}
            onPress={() => filterMovementsPressed(false)}>
            {labels.BUTTONS.REDEEMED}
          </RedeemedTotalButton>
        </HStack>
      </Box>
    );
  };

  const renderSingleButton = () => (
    <>
      <VStack>
        <TotalButton
          testID="total-points-button"
          bgColor="primary.100"
          _text={{
            fontSize: 16,
            fontWeight: 800,
            color: 'white',
            lineHeight: 22,
          }}
          onPress={() => allMovementsPressed()}>
          {labels.BUTTONS.ALL}
        </TotalButton>
      </VStack>
    </>
  );

  const renderMovementsList = () => (
    <FlatListContainer
      maxWidth="100%"
      bgColor={'white'}
      style={[styleHeight]}
      testID="movements-list-container">
      <FlatList
        data={currentMovements}
        renderItem={renderItem}
        keyExtractor={(item: Movement) => item.id.toString()}
      />
    </FlatListContainer>
  );

  const renderItem: ListRenderItem<Movement> = ({item}) => (
    <ProductItem movement={item} />
  );

  const renderLoadingList = () => (
    <HStack space={2} justifyContent="center">
      <Spinner accessibilityLabel="Loading" />
      <Heading color="primary.100" fontSize="md">
        {labels.LOADING.TITLE}
      </Heading>
    </HStack>
  );

  const render = () => (
    <SafeAreaView>
      <FullContainer>
        <Box border="1" borderRadius="md">
          <VStack>
            <WelcomeText color="text.title">
              {labels.WELCOME.MESSAGE}
            </WelcomeText>
          </VStack>
          <VStack>
            <WelcomeName color="text.title">{labels.WELCOME.NAME}</WelcomeName>
          </VStack>
        </Box>
        <Box>
          <Subtitle color="text.subtitle">{labels.SUBTITLES.POINTS}</Subtitle>
          <CardPadding>
            <CardContainer
              alignSelf="center"
              bg="primary.100"
              shadow={9}
              width="100%">
              <VStack>
                <CardMonth>{currentMonth}</CardMonth>
              </VStack>
              <VStack>
                <CardPointsTotal alignSelf="center">
                  {cardPoints}
                </CardPointsTotal>
              </VStack>
            </CardContainer>
          </CardPadding>
        </Box>
        <Box>
          <Subtitle color="text.subtitle">
            {labels.SUBTITLES.MOVEMENTS}
          </Subtitle>
          {loading ? renderLoadingList() : renderMovementsList()}
        </Box>
        <ButtonsContainer>
          {allMovements ? renderFilterButtons() : renderSingleButton()}
        </ButtonsContainer>
      </FullContainer>
    </SafeAreaView>
  );

  return render();
};

export default Movements;
