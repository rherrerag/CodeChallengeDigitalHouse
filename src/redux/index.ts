import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import {movementsReducer} from '../features/movements/reducers/movements.reducer';

export const store = configureStore({
  reducer: {
    movements: movementsReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
