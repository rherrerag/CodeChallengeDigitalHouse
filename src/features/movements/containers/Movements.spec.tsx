import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  act,
} from '@testing-library/react-native';
import {responseOK} from '../../../../__mocks__/responseProducts';
import {Provider} from 'react-redux';
import Movements from './Movements';
import {store} from '../../../redux';
import mockFetch from 'jest-fetch-mock';
import {extendTheme, NativeBaseProvider} from 'native-base';
import {NavigationContainer} from '@react-navigation/native';

const theme = extendTheme({
  colors: {
    primary: {
      100: '#334FFA',
      200: '#CFD6FF',
    },
    text: {
      title: '#020202',
      subtitle: '#9B9898',
      normal: '#000000',
    },
  },
});

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

jest.mock('react-native/Libraries/Lists/FlatList', () => 'FlatList');

const MovementsFactory = async () => {
  mockFetch.mockResponse(req => {
    if (req.url.includes('products')) {
      return Promise.resolve(JSON.stringify(responseOK.data));
    }

    return Promise.reject(new Error('not mapped url'));
  });
  render(
    <Provider store={store}>
      <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
        <NavigationContainer>
          <Movements />
        </NavigationContainer>
      </NativeBaseProvider>
    </Provider>,
  );
  await waitFor(() =>
    expect(screen.getAllByTestId('movements-list-container')).toBeTruthy(),
  );
};

describe('Movements Container', () => {
  beforeEach(() => {
    mockFetch.enableMocks();
    mockFetch.resetMocks();
  });

  it('debería generar snapshot de container correctamente', async () => {
    await MovementsFactory();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('debería mostrar Marzo en el mes de la Card', async () => {
    await MovementsFactory();
    await act(() => {
      expect(screen.getByText('Marzo')).toBeTruthy();
    });
  });

  it('debería mostrar botón Todos al pulsar boton de Ganados', async () => {
    await MovementsFactory();
    await act(() => {
      fireEvent(screen.getByTestId('earned-points-button'), 'press');
    });
    expect(screen.getByTestId('total-points-button')).toBeTruthy();
  });

  it('debería mostrar botón Todos al pulsar boton de Canjeados', async () => {
    await MovementsFactory();
    await act(() => {
      fireEvent(screen.getByTestId('redeemed-points-button'), 'press');
    });
    expect(screen.getByTestId('total-points-button')).toBeTruthy();
  });
});
