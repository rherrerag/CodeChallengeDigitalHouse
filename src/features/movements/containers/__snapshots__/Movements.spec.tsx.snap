// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movements Container deber√≠a generar snapshot de container correctamente 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaView>
    <View
      dataSet={{}}
      style={
        [
          {},
          [
            {
              "flexDirection": "column",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        border="1"
        dataSet={{}}
        style={
          {
            "borderRadius": 6,
          }
        }
      >
        <View
          dataSet={{}}
          style={{}}
        >
          <Text
            dataSet={{}}
            style={
              [
                {
                  "backgroundColor": undefined,
                  "color": "#020202",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                },
                [
                  {
                    "fontSize": 20,
                    "fontWeight": "800",
                    "lineHeight": 27,
                  },
                  undefined,
                ],
              ]
            }
          >
            Bienvenido de Vuelta!
          </Text>
        </View>
        <View
          dataSet={{}}
          style={{}}
        >
          <Text
            dataSet={{}}
            style={
              [
                {
                  "backgroundColor": undefined,
                  "color": "#020202",
                  "fontFamily": undefined,
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textDecorationLine": undefined,
                },
                [
                  {
                    "fontSize": 16,
                    "lineHeight": 27,
                  },
                  undefined,
                ],
              ]
            }
          >
            Ruben Rodriguez
          </Text>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      >
        <Text
          dataSet={{}}
          style={
            [
              {
                "backgroundColor": undefined,
                "color": "#9B9898",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              },
              [
                {
                  "fontSize": 14,
                  "fontWeight": "800",
                  "lineHeight": 19,
                  "paddingBottom": 20,
                  "paddingTop": 20,
                },
                undefined,
              ],
            ]
          }
        >
          TUS PUNTOS
        </Text>
        <View
          dataSet={{}}
          style={
            [
              {},
              [
                {
                  "paddingLeft": 34,
                  "paddingRight": 34,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            dataSet={{}}
            style={
              [
                {
                  "alignSelf": "center",
                  "backgroundColor": "#334FFA",
                  "elevation": 10,
                  "shadowColor": "#000000",
                  "shadowOffset": {
                    "height": 5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.34,
                  "shadowRadius": 6.27,
                  "width": "100%",
                },
                [
                  {
                    "borderBottomLeftRadius": 20,
                    "borderBottomRightRadius": 20,
                    "borderTopLeftRadius": 20,
                    "borderTopRightRadius": 20,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <Text
                dataSet={{}}
                style={
                  [
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    },
                    [
                      {
                        "color": "white",
                        "fontSize": 16,
                        "fontWeight": "800",
                        "lineHeight": 22,
                        "paddingBottom": 7,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Marzo
              </Text>
            </View>
            <View
              dataSet={{}}
              style={{}}
            >
              <Text
                dataSet={{}}
                style={
                  [
                    {
                      "alignSelf": "center",
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    },
                    [
                      {
                        "color": "white",
                        "fontSize": 32,
                        "fontWeight": "800",
                        "lineHeight": 44,
                        "paddingBottom": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                41,673 pts
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        dataSet={{}}
        style={{}}
      >
        <Text
          dataSet={{}}
          style={
            [
              {
                "backgroundColor": undefined,
                "color": "#9B9898",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              },
              [
                {
                  "fontSize": 14,
                  "fontWeight": "800",
                  "lineHeight": 19,
                  "paddingBottom": 20,
                  "paddingTop": 20,
                },
                undefined,
              ],
            ]
          }
        >
          TUS MOVIMIENTOS
        </Text>
        <View
          dataSet={{}}
          style={
            [
              {
                "backgroundColor": "#FFFFFF",
                "maxWidth": "100%",
              },
              [
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "paddingBottom": 16,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 16,
                },
                [
                  {
                    "height": 533.6,
                  },
                ],
              ],
            ]
          }
          testID="movements-list-container"
        >
          <FlatList
            data={
              [
                {
                  "createdAt": "2023-03-11T04:50:50.205Z",
                  "id": "2",
                  "image": "https://loremflickr.com/640/480/technics",
                  "is_redemption": false,
                  "points": 92984,
                  "product": "Recycled Plastic Tuna",
                },
                {
                  "createdAt": "2023-03-11T04:35:35.259Z",
                  "id": "3",
                  "image": "https://loremflickr.com/640/480/technics",
                  "is_redemption": false,
                  "points": 42416,
                  "product": "Fantastic Granite Bacon",
                },
                {
                  "createdAt": "2023-03-11T13:57:33.428Z",
                  "id": "4",
                  "image": "https://loremflickr.com/640/480/city",
                  "is_redemption": true,
                  "points": 23913,
                  "product": "Fantastic Fresh Gloves",
                },
                {
                  "createdAt": "2023-03-11T12:17:47.118Z",
                  "id": "5",
                  "image": "https://loremflickr.com/640/480/people",
                  "is_redemption": true,
                  "points": 69814,
                  "product": "Rustic Rubber Bacon",
                },
              ]
            }
            keyExtractor={[Function]}
            renderItem={[Function]}
          />
        </View>
      </View>
      <View
        dataSet={{}}
        style={
          [
            {},
            [
              {
                "justifyContent": "flex-end",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          dataSet={{}}
          style={{}}
        >
          <View
            dataSet={{}}
            style={
              {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#334FFA",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  },
                  [
                    {
                      "borderBottomLeftRadius": 10,
                      "borderBottomRightRadius": 10,
                      "borderTopLeftRadius": 10,
                      "borderTopRightRadius": 10,
                      "height": 50,
                      "marginBottom": 40,
                      "marginRight": 13,
                      "marginTop": 20,
                      "width": "48%",
                    },
                    undefined,
                  ],
                ]
              }
              testID="earned-points-button"
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#FFFFFF",
                        "fontFamily": undefined,
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "800",
                        "letterSpacing": 0,
                        "lineHeight": 16,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Ganados
                  </Text>
                </View>
              </View>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "width": 8,
                }
              }
            />
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#334FFA",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  },
                  [
                    {
                      "borderBottomLeftRadius": 10,
                      "borderBottomRightRadius": 10,
                      "borderTopLeftRadius": 10,
                      "borderTopRightRadius": 10,
                      "height": 50,
                      "marginBottom": 40,
                      "marginTop": 20,
                      "width": "48%",
                    },
                    undefined,
                  ],
                ]
              }
              testID="redeemed-points-button"
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#FFFFFF",
                        "fontFamily": undefined,
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "800",
                        "letterSpacing": 0,
                        "lineHeight": 16,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Canjeados
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaProvider>
`;
